src/modules/tests/test-data/cat-adi-part2/vehicle-checks/vehicle-checks.cat-adi-part2.reducer.ts

```diff
@@ -34,6 +72,54 @@ export function vehicleChecksCatADI2Reducer(
         ...state,
         showMeTellMeComments: action.comment,
       };
+    case vehicleChecksCatADI2ActionTypes.SHOW_ME_QUESTION_DRIVING_FAULT:
+      showMeQuestionIndex = newShowMeQuestionsState.findIndex(e => e.outcome === CompetencyOutcome.P);
+      if (showMeQuestionIndex > -1 && showMeTellMeFaultCount < 4) {
+        newShowMeQuestionsState[showMeQuestionIndex].outcome = CompetencyOutcome.DF;
+      }
+      return {
+        ...state,
+        showMeQuestions: newShowMeQuestionsState,
+      };
+    case vehicleChecksCatADI2ActionTypes.VEHICLE_CHECKS_SERIOUS_FAULT:
+      let dangerousFlag: boolean = false;
+      if (state.dangerousFault) {
+        dangerousFlag = true;
+      }
+      return {
+        ...state,
+        seriousFault: true,
+        dangerousFault: dangerousFlag,
+      };
+    case vehicleChecksCatADI2ActionTypes.VEHICLE_CHECKS_DANGEROUS_FAULT:
+      let seriousFlag: boolean = false;
+      if (state.seriousFault) {
+        seriousFlag = true;
+      }
+      return {
+        ...state,
+        seriousFault: seriousFlag,
+        dangerousFault: true,
+      };
+    case vehicleChecksCatADI2ActionTypes.VEHICLE_CHECKS_REMOVE_SERIOUS_FAULT:
+      return {
+        ...state,
+        seriousFault: false,
+      };
+    case vehicleChecksCatADI2ActionTypes.VEHICLE_CHECKS_REMOVE_DANGEROUS_FAULT:
+      return {
+        ...state,
+        dangerousFault: false,
+      };
+    case vehicleChecksCatADI2ActionTypes.SHOW_ME_QUESTION_PASSED:
+      showMeQuestionIndex = state.showMeQuestions.findIndex(e => e.outcome === CompetencyOutcome.DF);
```

Comment:
is this functionality trying to say when outcome is a DF then set it as a P?

if so think we should stick to using map like in other reducers, something like:

showMeQuestions:  state.showMeQuestions.map(question => ({...question, outcome: question.outcome === CompetencyOutcome.DF ? CompetencyOutcome.P : question.outcome}));

this also happens above so might be worth converting the one on line 76 also

Reply:
SHOW_ME_QUESTION_PASSED fires when you remove a DF from the tellme/showme button.  If there's less then 2 DF it inserts a P to make sure that there's always two records.

the one on line 76 (SHOW_ME_QUESTION_DRIVING_FAULT) does the opposite and fires when you add a DF, and prevents more than 4 faults of either kind.

Happy to look to combine, but dont see why they shouldnt be seperate.

Reply:
i didnt say they should be separate, i meant to use the map logic for both of the pieces of code instead of find index, all reducers using map so pointing it out for consistency

Reply:
ive re-written them slightly to address one of your other comments. i dont feel comfatable re-righting them completely to use the map function as the're working.  Dont let that stop you if you want to tho.

Reply:
maybe we should all have a tech discussion about it? im only trying to point it out as i cant see actual logic happening in many other reducers

Reply:
re-written